### DISCUSSION TITLE
Speeding up azure CI

### DISCUSSION THREAD

**Discussion by renefritze** [pymor/pymor#1222] [Ideas]
Currently we test all 4 supported pythons on both OSX and Windows. We're limited to 6 (or 8?) parallel jobs. A single job takes anywhere between ~27 (osx py 3.6) to 65 min (win py 3.9). In turn a full suite (PR + Push) can take 3 hours.

One thing we need to to is investigate why there's such a big difference in runtime between essentially the same jobs, and potentially speed up single runs.

But most importantly I think we need to reduce the amount of operating system + Python versions. 

What do you guys think?

**Comment by renefritze** [comment]
> But most importantly I think we need to reduce the amount of operating system + Python versions.
> 
> What do you guys think?

pinging @pymor/all

**Reply by sdrave** [reply]
Maybe only test the oldest and newest Python?

**Reply by renefritze** [reply]
I'll get on this on Monday if nobody voices dissent till then.
